RecoTauTag/HLTProducers/src/TauTagFilter.cc

```diff
@@ -0,0 +1,156 @@
+        minPt_(cfg.getParameter<double>("minPt")),
+        maxEta_(cfg.getParameter<double>("maxEta")),
+        usePtCorr_(cfg.getParameter<bool>("usePtCorr")),
+        matchWithSeeds_(cfg.getParameter<bool>("matchWithSeeds")),
+        matchingdR2_(std::pow(cfg.getParameter<double>("matchingdR"), 2)) {}
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+    edm::ParameterSetDescription desc;
+    makeHLTFilterDescription(desc);
+    desc.add<int>("nExpected", 2)->setComment("number of expected taus per event");
+    desc.add<edm::InputTag>("taus", edm::InputTag(""))->setComment("input collection of taus");
+    desc.add<edm::InputTag>("tauTags", edm::InputTag(""))->setComment("input collection of tau tagger scores");
+    desc.add<edm::InputTag>("tauPtCorr", edm::InputTag(""))
+        ->setComment("input collection of multiplicative tau pt corrections");
+    desc.add<edm::InputTag>("seeds", edm::InputTag(""))->setComment("input collection of seeds");
+    desc.add<std::vector<int>>("seedTypes",
+                               {trigger::TriggerL1Tau, trigger::TriggerL1Jet, trigger::TriggerTau, trigger::TriggerJet})
+        ->setComment("list of seed object types");
+    desc.add<std::string>("selection", "0")->setComment("selection formula");
+    desc.add<double>("minPt", 20)->setComment("minimal tau pt");
+    desc.add<double>("maxEta", 2.5)->setComment("maximal tau abs(eta)");
+    desc.add<bool>("usePtCorr", false)->setComment("use multiplicative tau pt corrections");
+    desc.add<bool>("matchWithSeeds", false)->setComment("apply match with seeds");
+    desc.add<double>("matchingdR", 0.5)->setComment("deltaR for matching with seeds");
+    descriptions.addWithDefaultLabel(desc);
+  }
+
+  bool hltFilter(edm::Event& event,
+                 const edm::EventSetup& eventsetup,
+                 trigger::TriggerFilterObjectWithRefs& filterproduct) const override {
+    if (saveTags())
+      filterproduct.addCollectionTag(tausSrc_);
+
+    int nTauPassed = 0;
+
+    const auto tausHandle = event.getHandle(tausToken_);
+    const auto& taus = *tausHandle;
+
+    std::vector<LorentzVectorM> seed_p4s;
+    if (matchWithSeeds_) {
+      const auto& seeds = event.get(seedsSrc_);
+      for (const int seedType : seedTypes_) {
+        if (seedType == trigger::TriggerL1Tau) {
+          extractMomenta<l1t::TauVectorRef>(seeds, seedType, seed_p4s);
+        } else if (seedType == trigger::TriggerL1Jet) {
+          extractMomenta<l1t::JetVectorRef>(seeds, seedType, seed_p4s);
+        } else if (seedType == trigger::TriggerTau) {
+          extractMomenta<std::vector<reco::PFTauRef>>(seeds, seedType, seed_p4s);
+        } else if (seedType == trigger::TriggerJet) {
+          extractMomenta<std::vector<reco::PFJetRef>>(seeds, seedType, seed_p4s);
+        } else
+          throw cms::Exception("Invalid seed type", "PNetTauTagFilter::hltFilter") << "Invalid seed type: " << seedType;
```

Comment:
Is not better to throw an exception already at C-tor if there is a seed type which is not supported?